src/OpenTK.Mathematics/Data/Color4.cs

```diff
@@ -149,6 +152,44 @@ public int ToArgb()
                 (int)(color.B * byte.MaxValue));
         }
 
+        /// <summary>
+        /// Returns this Color4 as a Vector4.
+        /// </summary>
+        /// <param name="c">The Color4 to convert.</param>
+        /// <returns>The Color4 to convert into a Vector4.</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static explicit operator Vector4(Color4 c)
+        {
+            unsafe
+            {
+                return *((Vector4*)&c);
+            }
+        }
+
+        /// <summary>
+        /// Returns a new Vector4.
+        /// </summary>
+        /// <returns>f.</returns>
+        public Vector4 AsVector4()
+        {
+            return new Vector4(R, G, B, A);
+        }
+
+        /// <summary>
+        /// this is a bad idea.
+        /// </summary>
+        /// <returns>why is this necessary for simple operations.</returns>
+        public Vector4 AsVector4Unsafe()
+        {
+            unsafe
+            {
+                fixed (Color4* ptr = &this)
```

Comment:
I forgot to update this from the old implementation I initially suggested in #833, I'll change this to the version from Vector4 in the next commit.

Reply:
We still don't actually know if the pointer-cast is faster. That's the only thing.

Reckon you could run your benchmark with this version too?

Reply:
Will try a benchmark in just a moment.

Reply:
Great! Excited to see what the results are.

Reply:
Just finished it. Results:

Debug: 444 milliseconds
Release: 76 milliseconds

Slightly slower than raw pointer casting when in release mode, but not by a ton.

Reply:
Sounds good.

Reply:
the other 76ms is measurement with fixed and this pointer referencing. Whereas explicit operator yields

> Debug: 363 milliseconds
> Release: 38 milliseconds

Btw. is it possible to use the explicit operator to prevent usage of fixed as well.
And because it get's inlined it probably does even prevent the stack duplication and therefore should yield the same speed. Otherwise probably still faster even with 'dup'.